controllers/core/openstackcontrolplane_controller.go

```diff
@@ -245,6 +260,59 @@ func (r *OpenStackControlPlaneReconciler) reconcileNormal(ctx context.Context, i
 	return ctrl.Result{}, nil
 }
 
+func (r *OpenStackControlPlaneReconciler) reconcileDelete(ctx context.Context, instance *corev1beta1.OpenStackControlPlane, helper *helper.Helper) (ctrl.Result, error) {
+	var err error
+	overallRes := ctrl.Result{}
+
+	//
+	// Ensure a controlled teardown of the control plane...
+	//
+
+	// Ordered list of resource types to delete
+	// Each outer index of resource groupings must be fully deleted before the next index is considered
+	steps := [][]client.ObjectList{
+		{
+			&cinderv1.CinderList{},
+			&glancev1.GlanceList{},
+			&neutronv1.NeutronAPIList{},
+			&novav1.NovaList{},
+			&ovnv1.OVNDBClusterList{},
+			&ovnv1.OVNNorthdList{},
+			&ovsv1.OVSList{},
+			&placementv1.PlacementAPIList{},
+		},
+		{
+			&rabbitmqv1.RabbitmqClusterList{},
+			&keystonev1.KeystoneAPIList{},
+		},
+		{
+			&mariadbv1.MariaDBList{},
+		},
+	}
+
+	for _, step := range steps {
+		if overallRes, err = openstack.CheckDeleteSubresources(ctx, helper, step); err != nil {
+			return overallRes, err
+		}
+
+		// If we're still waiting for this step's resources to be deleted, stop here
+		if (overallRes != ctrl.Result{}) {
+			util.LogForObject(helper, fmt.Sprintf("OpenStackControlPlane %s deletion is waiting on sub-resource deletion, requeuing...", instance.Name), instance)
+			return overallRes, nil
+		}
+	}
+
+	// Everything is cleared, so remove the finalizer so this OpenStackControlPlane can be fully removed
+	controllerutil.RemoveFinalizer(instance, helper.GetFinalizer())
+	if err := r.Update(ctx, instance); err != nil && !k8s_errors.IsNotFound(err) {
+		return overallRes, err
+	}
+
+	r.Log.Info("Reconciled Service delete successfully")
```

Comment:
Should this say 'ControlPlane' instead of service (sort of a collection of services

Reply:
Yeah, we can do something more specific.  I will change it.